MDCStatefulRipple.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCRippleView.h"
+
+/**
+ Provides the current state of the ripple. The ripple is either in its normal state, or in the
+ selected state where the ripple remains spread on the view.
+ - MDCRippleStateNormal: The ripple isn't currently presented.
+ - MDCRippleStateHighlighted: The ripple is activated and shown.
+ - MDCRippleStateSelected: The ripple is in the selected state.
+ - MDCRippleStateDragged: The ripple is in the dragged state.
+ */
+typedef NS_OPTIONS(NSInteger, MDCRippleStateNew) {
+  MDCRippleStateNewNormal = 0,
+  MDCRippleStateNewHighlighted = 1 << 0,
+  MDCRippleStateNewSelected = 1 << 1,
+  MDCRippleStateNewDragged = 1 << 2,
+};
+
+@interface MDCStatefulRippleView : MDCRippleView
+
+/**
+ This BOOL is set to YES if the ripple is currently selected, or NO otherwise.
+ It only has significance if selectionMode is activated.
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isSelected) BOOL selected;
+
+/**
+ This BOOL is set to YES if the ripple is currently highlighted, or NO otherwise.
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isRippleHighlighted) BOOL rippleHighlighted;
```

Comment:
Material theming uses "pressed" for the highlighted state".  We should make a decision about which naming conventions we should follow - material,  ios, material+ios, etc - so naming is consistent across components and schemes.

Reply:
Agreed with rob here, UIKit has used highlighted, selected, and dragged as its standard notions, such as UICollectionViewCells and UIControls

Reply:
Look at the design doc under Other Considerations -> UICollectionViewCell support and I will bring it up also in the meeting.

Reply:
@yarneo  @codeman7 It might be helpful to add a small (a few sentences?) inline comment to explain why we landed on this naming.  Since our internal design docs aren't public and they may get lost in the future, it's helpful to have a short note to explain decisions like this for future digital archaeologists.

Reply:
TLDR: internal implementation of UIKit for cells when selected changes all its subviews to highlighted = YES. We don't want the ripple view to receive that as it is supposed to imitate the cell's state not its subview's state.